Source/Core/DolphinQt2/ToolBar.cpp

```diff
@@ -91,15 +92,19 @@ void ToolBar::MakeActions()
 
 void ToolBar::UpdateIcons()
 {
-  QString dir = Settings().GetThemeDir();
-
-  m_open_action->setIcon(QIcon(QStringLiteral("open.png").prepend(dir)));
-  m_paths_action->setIcon(QIcon(QStringLiteral("browse.png").prepend(dir)));
-  m_play_action->setIcon(QIcon(QStringLiteral("play.png").prepend(dir)));
-  m_pause_action->setIcon(QIcon(QStringLiteral("pause.png").prepend(dir)));
-  m_stop_action->setIcon(QIcon(QStringLiteral("stop.png").prepend(dir)));
-  m_fullscreen_action->setIcon(QIcon(QStringLiteral("fullscreen.png").prepend(dir)));
-  m_screenshot_action->setIcon(QIcon(QStringLiteral("screenshot.png").prepend(dir)));
-  m_config_action->setIcon(QIcon(QStringLiteral("config.png").prepend(dir)));
-  m_controllers_action->setIcon(QIcon(QStringLiteral("classic.png").prepend(dir)));
+  m_open_action->setIcon(QIcon(Resources::GetScaledPixmap("open", 4, Resources::Directory::Theme)));
```

Comment:
Here's the issue: if you ever want to display these icons on a @2x screen at half their native size, e.g. 16x16, the appropriate strategy is to treat the @1x icons as actually 16x16@2x icons rather than downscale the 32x32@2x icons. This is the default behavior of QIcon, which is designed for easy handling of different sizes/densities/states, but it's prevented by this PR.

If using @1x pixmaps is really no different than downscaling @2x pixmaps, then we should get rid of the @1x ones altogether and always use @2x. Otherwise, you can fix this by adding each scale pixmap to the QIcon manually with something like:

```c++
QIcon icon;
icon.addFile(icon_name + QStringLiteral(".png")); // also automatically picks up @2x

QPixmap pixmap4x(icon_name + QStringLiteral("@4x.png"));
if (!pixmap4x.isNull()) {
  pixmap4x.setDevicePixelRatio(4.0);
  icon.addPixmap(pixmap4x);
}
```